daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/CommandPreprocessor.scala

```diff
@@ -94,7 +94,7 @@ private[lf] final class CommandPreprocessor(compiledPackages: CompiledPackages)
     val (arg, argCids) = valueTranslator.unsafeTranslateValue(choiceArgType, argument)
     val (key, keyCids) = valueTranslator.unsafeTranslateValue(ckTtype, contractKey)
     keyCids.foreach { coid =>
-      fail(s"Unexpected contract id in key: $coid")
+      fail(s"Contract IDs are not supported in contract keys: $coid")
```

Comment:
Would it make sense to include the templateId as well, or will this be clear in the context of the error message?

Reply:
sounds sensible, added.